README.md

```diff
@@ -671,89 +718,111 @@ window.requestFileSystem(window.TEMPORARY, 5 * 1024 * 1024, function (fs) {
+After getting the file, copy the contents to a new file. The current DirectoryEntry object is already associated with the app cache.
 
-```
-function saveFile(dirEntry, srcImage, fileName) {
+```js
+function saveFile(dirEntry, fileData, fileName) {
 
     dirEntry.getFile(fileName, { create: true, exclusive: false }, function (fileEntry) {
 
-        writeFile(fileEntry, srcImage);
+        writeFile(fileEntry, fileData);
 
     }, onErrorCreateFile);
 }
 ```
 
-In writeFile, you pass in the DOM string URL as the dataObj and you will save that in the new file.
+In writeFile, you pass in the Blob object as the dataObj and you will save that in the new file.
+
+```js
+function writeFile(fileEntry, dataObj, isAppend) {
 
-```
-function writeFile(fileEntry, dataObj) {
     // Create a FileWriter object for our FileEntry (log.txt).
     fileEntry.createWriter(function (fileWriter) {
 
-        fileWriter.onwriteend = function (e) {
+        fileWriter.onwriteend = function() {
             console.log("Successful file write...");
-            readFile(fileEntry);
+            if (dataObj.type == "image/png") {
+                readBinaryFile(fileEntry);
+            }
+            else {
+                readFile(fileEntry);
+            }
         };
 
-        fileWriter.onerror = function (e) {
+        fileWriter.onerror = function() {
             console.log("Failed file write: " + e.toString());
         };
 
-        // If data object is not passed in,
-        // create a new Blob instead.
-        if (!dataObj) {
-            dataObj = new Blob(['some file data'], { type: 'text/plain' });
-        }
-
         fileWriter.write(dataObj);
     });
 }
 ```
 
-After writing to the file, read it and display it. These operations re-use the code that we showed you already in previous tasks, so theres nothing new there (see the previous sections). After reading the data, you can display the image using code like this.
+After writing to the file, read it and display it. You saved the image as binary data, so you can read it using FileReader.readAsArrayBuffer.
+
+```js
+function readBinaryFile(fileEntry) {
+
+    fileEntry.file(function (file) {
+        var reader = new FileReader();
+
+        reader.onloadend = function() {
 
+            console.log("Successful file write: " + this.result);
+            displayFileData(fileEntry.fullPath + ": " + this.result);
+
+            var blob = new Blob([new Uint8Array(this.result)], { type: "image/png" });
+            displayImage(blob);
+        };
+
+        reader.readAsArrayBuffer(file);
+
+    }, onErrorReadFile);
+}
 ```
-function displayImageData(fileData) {
 
+After reading the data, you can display the image using code like this. Use window.URL.createObjectURL to get a DOM string for the Blob image.
+
+```js
+function displayImage(blob) {
+
+    var img = new Image();
+    // Note: Use window.URL.revokeObjectURL when finished with image.
+    img.src = window.URL.createObjectURL(blob);
```

Comment:
I don't think you need to create this extra Image object. You can assign the result of createObjectURL(blob) directly to elem.src.